kroxylicious-filters/kroxylicious-simple-transform/src/main/java/io/kroxylicious/proxy/filter/simpletransform/Replacing.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.simpletransform;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+
+@Plugin(configType = Replacing.Config.class)
+public class Replacing implements ByteBufferTransformationFactory<Replacing.Config> {
+    public record Config(
+                         @JsonProperty String charset,
+                         @JsonProperty(required = true) String targetPattern,
+                         @JsonProperty String replacementValue,
+                         @JsonProperty String replaceFrom) {
```

Comment:
I noticed recently that Jackson has a built in NioPathDeserializer which knows how to deserial Paths. Using that would make the API document itself.